src/gt4sd/frameworks/gflownet/util.py

```diff
@@ -0,0 +1,147 @@
+#
+# MIT License
+#
+# Copyright (c) 2022 GT4SD team
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+import queue
+import threading
+
+import torch
+import torch.multiprocessing as mp
+import torch.nn as nn
+
+
+class MPModelPlaceholder:
+    """This class can be used as a Model in a worker process, and
+    translates calls to queries to the main process.
+    """
+
+    def __init__(self, in_queues, out_queues):
+        self.qs = in_queues, out_queues
+        self.device = torch.device("cpu")
+        self._is_init = False
+
+    def _check_init(self):
+        if self._is_init:
+            return
+        info = torch.utils.data.get_worker_info()
+        self.in_queue = self.qs[0][info.id]
+        self.out_queue = self.qs[1][info.id]
+        self._is_init = True
+
+    # TODO: make a generic method for this based on __getattr__
+    def logZ(self, *a):
```

Comment:
```suggestion
    def log_z(self, *a):
```
and consider adding timing where possible.

Reply:
This `logZ`, forward log-probability for a trajectory (similar to a partition function) is a relevant quantity in gflownet. 
I think it can make sense to have the capital here in case we want to use `z` for a single step later on.

Reply:
I would still avoid to "break" the code style as we associate camel notation only to class types and object types.
In case logZ always indicates a quantity/variable we should stick `log_z`.
Consider the case for the SMILES that when used as variable is `smiles`.